RHDEVDOCS-3057: Securing Webhooks with Event Listeners (#34882)
openshift/openshift-docs: cicd/pipelines/securing-webhooks-with-event-listeners.adoc

[id="securing-webhooks-with-event-listeners"]
= Securing webhooks with event listeners
include::modules/common-attributes.adoc[]
include::modules/pipelines-document-attributes.adoc[]

:context: securing-webhooks-with-event-listeners

toc::[]

> Triggers in {pipelines-title} support both HTTP (insecure) and HTTPS (secure HTTP) connections to the `Eventlistener` resource. With the secure HTTPS connection, you get an end-to-end secure connection within and outside the cluster. After you create a namespace, you can enable this secure HTTPS connection for the `Eventlistener` resource by adding the `operator.tekton.dev/enable-annotation=enabled` label to the namespace, and then creating a `Trigger` resource and a secured route using re-encrypt TLS termination.

Reviewer: > After you create a namespace, you can enable this secure HTTPS connection for the `Eventlistener` resource by adding the `operator.tekton.dev/enable-annotation=enabled` label to the namespace, and then creating a `Trigger` resource and a secured route using re-encrypt TLS termination.

Too long sentence. Consider breaking down, use active voice, and establish a clear narrative.

> When an OpenShift pipeline is installed through an operator a `tekton-operator-proxy-webhook-` pod will always be running. The webhook watches for labels on the namespace and the below annotation is set on the `Eventlistener` object, whenever one is found, which in turn creates secrets and required certificates:

Reviewer: When you install a {pipelines-title} using the operator, a `tekton-operator-proxy-webhook` pod always runs in the background. The webhook watches for labels in the namespace, and the `service.beta.openshift.io/serving-cert-secret-name=<secret_name>` annotation is set on the `EventListener` object, whenever one is found. This, in turn, creates secrets and required certificates.

COMMENT - Even with this rephrasing, I am not satisfied. The technical meaning of the sentences are not clear. Try using active voice as much as possible, with a clear subject-verb-object narrative.

[source,terminal]
----
service.beta.openshift.io/serving-cert-secret-name=<secret_name>
----

> You can further mount the created secret into the `Eventlistener` pod to secure the request.

Reviewer: In addition, you can mount the created secret into the `Eventlistener` pod to secure the request.

> include::modules/op-providing-secure-connection.adoc[leveloffset=+1]

Reviewer: Add a blank line between the consecutive includes.

[id="securing-webhooks-with-event-listeners"]
= Securing webhooks with event listeners
include::modules/common-attributes.adoc[]
include::modules/pipelines-document-attributes.adoc[]

:context: securing-webhooks-with-event-listeners

toc::[]

> Triggers in {pipelines-title} support both HTTP (insecure) and HTTPS (secure HTTP) connections to the `Eventlistener` resource. With the secure HTTPS connection, you get an end-to-end secure connection within and outside the cluster. Enable the secure HTTPS connection, after creating a namespace, for the `Eventlistener` resource by adding the `operator.tekton.dev/enable-annotation=enabled` label to the namespace. After that, create a `Trigger` resource and a secured route using the re-encrypted TLS termination.

Reviewer 2: ```suggestion
Triggers in {pipelines-title} support insecure HTTP and secure HTTPS connections to the `Eventlistener` resource. HTTPS secures connections within and outside the cluster. After creating a namespace, you enable HTTPS for the `Eventlistener` resource by adding the `operator.tekton.dev/enable-annotation=enabled` label to the namespace. After that, you create a `Trigger` resource and a secured route using the re-encrypted TLS termination.
```

Reviewer 2: Consider dropping or modifying line 12 if you think this knowledge is already apparent to your users.

> When you install a {pipelines-title} using the operator, a `tekton-operator-proxy-webhook` pod always runs in the background. The webhook watches for the labels in the namespace, and the `service.beta.openshift.io/serving-cert-secret-name=<secret_name>` annotation is set on the `EventListener` object, whenever one is found. This, in turn, creates secrets and required certificates.

Reviewer 3: operator --> Operator

Reviewer: @HarshCasper: Sri is right - the rule is written here - https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#operator-name-capitalization. Please make the changes she suggests.

> When you install a {pipelines-title} using the Operator, a `tekton-operator-proxy-webhook` pod always runs in the background. The webhook watches for the labels in the namespace, and the `service.beta.openshift.io/serving-cert-secret-name=<secret_name>` annotation is set on the `EventListener` object, whenever one is found. This, in turn, creates secrets and required certificates.

Reviewer 2: ```suggestion
Installing {pipelines-title} using the Operator runs a `tekton-operator-proxy-webhook` pod in the background. This webhook watches for the labels in the namespace. When you add the label to the namespace, the webhook sets the `service.beta.openshift.io/serving-cert-secret-name=<secret_name>` annotation on the `EventListener` object. This, in turn, creates secrets and required certificates.
```

Reviewer 2: Is it important to call out installing by using the Operator? Are there other methods of installing that don't run the webhook pod in the background?

Reviewer 4: it will be installed using Operator only without Operator webhook pod won't run
